src/Workspaces/Core/Portable/Formatting/Formatter.cs

```diff
@@ -281,7 +282,13 @@ internal static Task<IFormattingResult> GetFormattingResult(SyntaxNode node, IEn
         /// <param name="cancellationToken">An optional cancellation token.</param>
         /// <returns>The changes necessary to format the tree.</returns>
         public static IList<TextChange> GetFormattedTextChanges(SyntaxNode node, Workspace workspace, OptionSet options = null, CancellationToken cancellationToken = default)
-            => GetFormattedTextChangesAsync(node, workspace, options, cancellationToken).WaitAndGetResult(cancellationToken);
+        {
+            options = options ?? workspace.Options;
+            options = options.WithChangedOption(FormattingOptions.AllowConcurrent, false);
+            var resultTask = GetFormattedTextChangesAsync(node, workspace, options, cancellationToken);
+            Debug.Assert(resultTask.IsCompleted);
+            return resultTask.WaitAndGetResult_CanCallOnBackground(cancellationToken);
```

Comment:
@jasonmalinowski Please make sure to review the new pattern in this method.

Reply:
This seems like this deserves a comment.  Is there a concern here about blocking a thread on a parallelized algorithm?

Reply:
the one who calls this API should set the option, not the API itself set the option.

Reply:
> Is there a concern here about blocking a thread on a parallelized algorithm?

Yes, the analyzer runs in parallel on the thread pool. It would be easy for it to starve the system.

Reply:
So, i'm somewhat ok with this change.  But it would def need to be applied a bit uniformly.  We could also consider deprecating this guy.  It always worries me when we expose sync entrypoints that immediately block on the async version.  

It should also be a warning to us for future API development.  It's likely we'll need to default things to being async so we don't end up hurting ourselves here later.  A good example of that is around the discussions of Options.  Options present a synchronous API, but need to block on async work.  If we then make Options async, it virally hits entrypoints like this (which are currently fine being synchronous).  

If we had settled on these being async from teh start, we wouldn't have these problems cropping up over time as more core parts of our system become async.

Reply:
> But it would def need to be applied a bit uniformly. 

Implemented this for the rest of the synchronous `Formatter` methods. It's so ugly it makes me sad but implementing the synchronous form all-the-way-down was a huge amount of work and mostly duplicated code.

Reply:
so the thing i'm missing is: how do you know the returned task iscomplete?

Reply:
oh, i see. because you pass AllowConcurrent: false.  gotcha.

Reply:
Note: my preference wouldjust be deprecate these methods as we do this.

Reply:
The diagnostic analyzer code paths are synchronous, and rely on the new synchronous behavior of these methods to avoid thread pool starvation.